From 8140146d56ba5833b2e45db3cafe6737bf1ec4b4 Mon Sep 17 00:00:00 2001
From: Yassin <yassin@Nixos>
Date: Mon, 24 Aug 2026 16:03:09 +0200
Subject: [PATCH 6/9] arm64: dts: mediatek: add Lenovo Tab M10 FHD Plus

Add the Lenovo TB-X606F (achilles6_row_wifi), with its verified 4 GiB
memory map, persistent RAM regions, eMMC wiring, MT6357 interrupt and
USB peripheral mode.
---
 arch/arm64/boot/dts/mediatek/Makefile         |   1 +
 .../dts/mediatek/mt8768-lenovo-tb-x606f.dts   | 102 ++++++++++++++++++
 2 files changed, 103 insertions(+)
 create mode 100644 arch/arm64/boot/dts/mediatek/mt8768-lenovo-tb-x606f.dts

diff --git a/arch/arm64/boot/dts/mediatek/Makefile b/arch/arm64/boot/dts/mediatek/Makefile
index a86fb313b..e3e01a77e 100644
--- a/arch/arm64/boot/dts/mediatek/Makefile
+++ b/arch/arm64/boot/dts/mediatek/Makefile
@@ -179,6 +179,7 @@ dtb-$(CONFIG_ARCH_MEDIATEK) += mt8395-radxa-nio-12l-8-hd-panel.dtbo
 mt8395-radxa-nio-12l-8-hd-panel-dtbs := mt8395-radxa-nio-12l.dtb mt8395-radxa-nio-12l-8-hd-panel.dtbo
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8395-radxa-nio-12l-8-hd-panel.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt8516-pumpkin.dtb
+dtb-$(CONFIG_ARCH_MEDIATEK) += mt8768-lenovo-tb-x606f.dtb
 
 # Device tree overlays support
 DTC_FLAGS_mt7986a-bananapi-bpi-r3 := -@
diff --git a/arch/arm64/boot/dts/mediatek/mt8768-lenovo-tb-x606f.dts b/arch/arm64/boot/dts/mediatek/mt8768-lenovo-tb-x606f.dts
new file mode 100644
index 000000000..a43182ae2
--- /dev/null
+++ b/arch/arm64/boot/dts/mediatek/mt8768-lenovo-tb-x606f.dts
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/* Lenovo Tab M10 FHD Plus (TB-X606F, achilles6_row_wifi) */
+
+/dts-v1/;
+
+#include "mt6765.dtsi"
+#include "mt6357.dtsi"
+
+/ {
+	model = "Lenovo Tab M10 FHD Plus (TB-X606F)";
+	compatible = "lenovo,tb-x606f", "mediatek,mt6765";
+
+	memory@40000000 {
+		device_type = "memory";
+		reg = <0 0x40000000 1 0x00000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		ram-console@47c80000 {
+			reg = <0 0x47c80000 0 0x10000>;
+			no-map;
+		};
+
+		ramoops@47c90000 {
+			compatible = "ramoops";
+			reg = <0 0x47c90000 0 0xe0000>;
+			record-size = <0x40000>;
+			console-size = <0x40000>;
+			pmsg-size = <0x10000>;
+		};
+
+		minidump@47d70000 {
+			reg = <0 0x47d70000 0 0x10000>;
+			no-map;
+		};
+	};
+};
+
+&mmc0 {
+	bus-width = <8>;
+	max-frequency = <52000000>;
+	cap-mmc-highspeed;
+	cap-mmc-hw-reset;
+	non-removable;
+	no-sd;
+	no-sdio;
+	pinctrl-names = "default", "state_uhs";
+	pinctrl-0 = <&mmc0_pins>;
+	pinctrl-1 = <&mmc0_pins>;
+	vmmc-supply = <&mt6357_vemc_reg>;
+	vqmmc-supply = <&mt6357_vio18_reg>;
+	status = "okay";
+};
+
+&pio {
+	mmc0_pins: mmc0-pins {
+		pins-cmd-dat-rst {
+			pinmux = <(MTK_PIN_NO(122) | 1)>,
+				 <(MTK_PIN_NO(123) | 1)>,
+				 <(MTK_PIN_NO(125) | 1)>,
+				 <(MTK_PIN_NO(126) | 1)>,
+				 <(MTK_PIN_NO(127) | 1)>,
+				 <(MTK_PIN_NO(128) | 1)>,
+				 <(MTK_PIN_NO(129) | 1)>,
+				 <(MTK_PIN_NO(130) | 1)>,
+				 <(MTK_PIN_NO(132) | 1)>,
+				 <(MTK_PIN_NO(133) | 1)>;
+			input-enable;
+			drive-strength = <3>;
+			bias-pull-up = <MTK_PUPD_SET_R1R0_01>;
+		};
+
+		pins-clk-dsl {
+			pinmux = <(MTK_PIN_NO(124) | 1)>,
+				 <(MTK_PIN_NO(131) | 1)>;
+			drive-strength = <3>;
+			bias-pull-down = <MTK_PUPD_SET_R1R0_10>;
+		};
+	};
+};
+
+&mt6357_pmic {
+	interrupts-extended = <&pio 144 IRQ_TYPE_LEVEL_HIGH>;
+	interrupt-controller;
+	#interrupt-cells = <2>;
+};
+
+&pwrap {
+	status = "okay";
+};
+
+&u2phy {
+	status = "okay";
+};
+
+&usb2 {
+	status = "okay";
+};
